src/modules/shared/components/Navigation/NavigationList/NavigationList.tsx

```diff
@@ -1,14 +1,80 @@
+import { Badge } from '@meemoo/react-components';
 import clsx from 'clsx';
 import Link from 'next/link';
-import { FC } from 'react';
+import { FC, ReactNode, useState } from 'react';
 
+import { Icon, IconLightNames, Overlay } from '@shared/components';
+
+import { NavigationDropdownItem, NavigationItem } from '..';
 import styles from '../Navigation.module.scss';
+import { NavigationDropdown } from '../NavigationDropdown';
 
 import { NavigationListProps } from './NavigationList.types';
 
 const NavigationList: FC<NavigationListProps> = ({ items }) => {
+	const [openDropdown, setOpenDropdown] = useState('');
+
+	const renderLink = (
+		item: NavigationItem,
+		linkCls: string,
+		icon: IconLightNames | undefined = undefined
+	): ReactNode => {
+		return (
+			<Link href={item.href}>
+				<a className={linkCls}>
+					{item.label}
+					<span className={styles['c-navigation__border-decoration']} />
+					<span className={styles['c-navigation__border-decoration']} />
+					{item.badge && <Badge text={item.badge} />}
+					{icon && <Icon className="u-text-left u-ml-4" name={icon} />}
+				</a>
+			</Link>
+		);
+	};
+
+	const renderDropdown = (
+		id: string,
+		trigger: ReactNode,
+		dropdownItems: NavigationItem[][]
+	): ReactNode => {
+		const desktopDropdownItems = dropdownItems.map((menuItems) => {
+			return menuItems.filter((item: NavigationDropdownItem) => item.showOnlyOn !== 'mobile');
+		});
+
+		return (
+			<>
```

Comment:
Moet die nog gewrapped worden in een fragment?

Reply:
Nope, was nog van toen de dropdown niet apart stond.